calico_containers/calico_ctl/bgp.py

```diff
@@ -54,10 +54,16 @@ def validate_arguments(arguments):
                     validate_ip(arguments["<PEER_IP>"], 4) or \
                     validate_ip(arguments["<PEER_IP>"], 6)
     asnum_ok = True
-    if arguments.get("<AS_NUM>"):
+    if arguments.get("<AS_NUM>") or arguments.get("--as"):
         try:
-            asnum = int(arguments["<AS_NUM>"])
-            asnum_ok = 0 <= asnum <= 4294967295
+            asnum = str(arguments.get("<AS_NUM>") or arguments.get("--as"))
+            if "." in asnum:
+                left_asnum, right_asnum = asnum.split(".")
+                if not (0 <= (int(left_asnum) | int(right_asnum)) <= 65535):
+                    raise ValueError
+                asnum_ok = 0 <= (65536*int(left_asnum)+int(right_asnum)) <= 4294967295
```

Comment:
This is pretty much testing the same thing as you meant to test in the if statement above: `65535.65535 == 4294967295`

You can just set `asnum_ok = <if statement logic>`